blueocean-plugin/src/main/resources/archetype-resources/src/main/java/Computers.java

```diff
@@ -0,0 +1,163 @@
+        return organization;
+    }
+    
+    @Exported(inline=true)
+    public ComputerInfo[] getComputers() throws Exception {
+        List<ComputerInfo> info = new ArrayList<>();
+        Jenkins j = Jenkins.getInstance();
+        for (Computer c : j.getComputers()) {
+            info.add(new ComputerInfo(getLink(), c));
+        }
+        return info.toArray(new ComputerInfo[info.size()]);
+    }
+
+    @ExportedBean
+    public static class ComputerInfo extends Resource {
+        final Link parent;
+        Computer computer;
+
+        public ComputerInfo(Link parent, Computer computer) {
+            this.parent = parent;
+            this.computer = computer;
+        }
+
+        @Exported
+        public String getDisplayName() {
+            return computer.getDisplayName();
+        }
+
+        @Exported(inline=true)
+        public Container<ExecutorInfo> getExecutors() {
+            final List<ExecutorInfo> out = new ArrayList<>();
+            if (computer != null) {
+                for (Executor e : computer.getExecutors()) {
+                    out.add(new ExecutorInfo(this, e));
+                }
+            }
+            return new Container() {
+                @Override
+                public Object get(String string) {
+                    throw new UnsupportedOperationException("Not supported.");
+                }
+
+                @Override
+                public Iterator iterator() {
+                    return out.iterator();
+                }
+
+                @Override
+                public Link getLink() {
+                    return parent.rel("executors");
+                }
+            };
+        }
+
+        @Override
+        public Link getLink() {
+            return parent.rel("computer");
+        }
+    }
+    
+    @ExportedBean
+    public static class ExecutorInfo extends Resource {
+        final Resource parent;
+        final Executor executor;
+        private ExecutorInfo(Resource parent, Executor executor) {
+            this.parent = parent;
+            this.executor = executor;
+        }
+
+        @Override
+        public Link getLink() {
+            return parent.getLink().rel("executor");
+        }
+        
+        @Exported
+        public String getDisplayName() {
+            return executor.getDisplayName();
+        }
+        
+        @Exported
+        public boolean isIdle() {
+            return executor.isIdle();
+        }
+
+        @Exported(inline=true)
+        public BlueRun getRun() {
+            Queue.Executable e = executor.getCurrentExecutable();
+            if (e == null) {
+                return null;
+            }
+            Run r = null;
+            SubTask subTask = e.getParent();
+            if (subTask instanceof ExecutorStepExecution.PlaceholderTask) {
+                ExecutorStepExecution.PlaceholderTask task = (ExecutorStepExecution.PlaceholderTask)subTask;
+                r = task.run();
```

Comment:
This is not intended as an API and I would not want to advertise to people that they should declare a dep on `workflow-durable-task-step` at all.

IIUC you are trying to go from `PlaceholderExecutable` to `Run & Executable`. This would argue for a new API method in `Executable` like

```java
default @CheckForNull Executable getOwnerExecutable() {
    return null;
}
```

which `PlaceholderExecutable` would implement by delegating to `PlaceholderTask.run()`.

In the short term, to avoid requiring a recent core dep, `workflow-durable-task-step` could implement the method without `@Override`, and the example client could check for a method of that signature reflectively.

But anyway for purposes of an archetype I would suggest something simpler than this. People are going to want to delete all this code anyway and replace with their own stuff—we just need to show them how to wire things up into the BO UI.